internal/k8s/exec.go

```diff
@@ -160,50 +163,49 @@ func podExec(kubeConfigPath string, ps *PodSessionOptions, pe v1.PodExecOptions,
 	return nil
 }
 
-// ReturnDeploymentObject returns a matching appsv1.Deployment object based on the filters
-func ReturnDeploymentObject(clientset *kubernetes.Clientset, matchLabel string, matchLabelValue string, namespace string, timeoutSeconds int) (*appsv1.Deployment, error) {
+func ReturnDeploymentObject(client kubernetes.Interface, matchLabel string, matchLabelValue string, namespace string, timeoutSeconds int) (*appsv1.Deployment, error) {
 
-	// Filter
-	deploymentListOptions := metav1.ListOptions{
-		LabelSelector: fmt.Sprintf("%s=%s", matchLabel, matchLabelValue),
+	timeout := time.Duration(timeoutSeconds) * time.Second
+	var deployment *appsv1.Deployment
+	labelSelector := map[string]string{
+		matchLabel: matchLabelValue,
 	}
 
-	log.Info().Msgf("waiting for %s Deployment to be created", matchLabelValue)
+	err := wait.PollImmediate(15*time.Second, timeout, func() (bool, error) {
+		deployments, err := client.AppsV1().Deployments(namespace).List(context.Background(), metav1.ListOptions{
+			LabelSelector: metav1.FormatLabelSelector(metav1.SetAsLabelSelector(labelSelector)),
+		})
```

Comment:
I see you merged our two plans: passing labels as a single map vs generating them here.

For the sake of simplicity, I would:

```go
	timeout := time.Duration(timeoutSeconds) * time.Second
	var deployment *appsv1.Deployment

	err := wait.PollImmediate(15*time.Second, timeout, func() (bool, error) {
		deployments, err := client.AppsV1().Deployments(namespace).List(context.Background(), metav1.ListOptions{
			LabelSelector: fmt.Sprintf("%s=%s", matchLabel, matchLabelValue),
		})
```

It's less lines of code 🤣

Reply:
Done :)